static/js/stream_popover.js

```diff
@@ -486,6 +486,10 @@ function build_move_topic_to_stream_popover(e, current_stream_id, topic_name) {
             default_text: $t({defaultMessage: "No streams"}),
             include_current_item: false,
             value: current_stream_id,
+            on_update: (selected_stream_id) => {
+                $("#move_topic_modal .dialog_submit_button")[0].disabled =
+                    Number.parseInt(selected_stream_id, 10) === current_stream_id;
+            },
```

Comment:
Hmm, what happens if you've edited one of the other inputs, like the destination topic?

Reply:
Oh, I see you fixed that in the later commits. You should generally squash fixup changes into the commit that introduces the problem you're fixing -- PRs with multiple commits are intended for a different purpose, which is to make a series of atomic changes that are each individually correct and easy to verify but would be hard to understand if squashed together.

Check out our [GitHub guide](https://zulip.readthedocs.io/en/latest/git-guide.html) and [commit guidelines](https://zulip.readthedocs.io/en/latest/contributing/version-control.html) for more details.

Reply:
Oh, thanks. Definitely wasn't 100% sure if I should squash these, will do from now on.